packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx

```diff
@@ -41,35 +45,98 @@ function splitSpaceName(
   return [start, remainder, end];
 }
 
-type NavbarSwitcherOwnProps = CommonProps & {
-  children: ReactNode;
-  isCircle?: boolean;
-  envVariant?: 'master' | 'non-master';
-  isAlias?: boolean;
-  ariaLabel?: string;
-};
+type NavbarLoadingProps =
+  | {
+      isLoading?: true;
+      environemnt?: never;
+      space?: never;
+    }
+  | {
+      isLoading?: false;
+      environemnt?: string;
+      space?: string;
+    };
+
+type NavbarSwitcherOwnProps = CommonProps &
+  NavbarLoadingProps & {
+    isCircle?: boolean;
+    envVariant?: 'master' | 'non-master';
+    isAlias?: boolean;
+    ariaLabel?: string;
+    environment?: React.ReactNode;
+    space?: React.ReactNode;
+  };
 
 export type NavbarSwitcherProps = PropsWithHTMLElement<
   NavbarSwitcherOwnProps,
   'button'
 >;
 
+type SwitcherLabelEnvVariantProps =
+  | {
+      type: 'environment';
+      envVariant?: 'master' | 'non-master';
+    }
+  | {
+      type: 'space';
+      envVariant?: never;
+    };
+
+type SwitcherLabelProps = SwitcherLabelEnvVariantProps & {
+  value: React.ReactNode;
+  styles: ReturnType<typeof getNavbarSwitcherStyles>;
+  type: 'space' | 'environment';
+  envVariant?: 'master' | 'non-master';
+};
+
+const SwitcherLabel = ({
+  value,
+  styles,
+  envVariant,
+  type,
+}: SwitcherLabelProps) => {
+  const [start, middle, end] =
+    typeof value === 'string' ? splitString(value) : [];
+  const envColor = useMemo(() => {
+    const isEnv = type === 'environment';
+    const isMaster = envVariant === 'master';
+    switch (true) {
+      case isEnv && isMaster:
+        return 'green600';
+      default:
+        return 'gray600';
+    }
```

Comment:
Can we do if statements and not try to be overly clever?

Reply:
Sorry @massao 😄